### PR TITLE
Adds txid and statsd modules. Refactors.

### PR DESCRIPTION
**Pull request by dansowter**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,32 +2,37 @@ FROM debian:wheezy
...
-  apt-get build-dep -y nginx=${NGINX_VERSION} && \
-  cd /tmp && apt-get source nginx=${NGINX_VERSION} && \
+  unzip
+
+RUN git clone https://github.com/streadway/ngx_txid.git /tmp/ngx_txid
+RUN git clone https://github.com/zebrafishlabs/nginx-statsd.git /tmp/nginx-statsd
+RUN apt-get build-dep -y nginx=${NGINX_VERSION}
+
+# TODO use a patch, rather than a replacement 'rules'
+COPY rules /tmp/rules
```
Comment (coop) [inline Dockerfile:24]: `diff -u`

File: update_changelog.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+nginx ($NGINX_VERSION+edh) wheezy; urgency=low
+  * Update to upstream version $NGINX_VERSION
+  * Includes custom EDH modules.
+
+ -- Dan Sowter <daniel.sowter@everydayhero.com>  Tue, 17 Mar 2015 14:40:14 +1000
+
+EOF
+}
+
+cat <(new_changelog_entry) /tmp/${NGINX_SOURCE_FOLDER}/debian/changelog > temp_changelog
```
Comment (coop) [inline update_changelog.sh:14]: Geez, he's not mucking around here with his unix is he?

**Comment by dansowter** [comment]
Tried a nastier single RUN command version to see what the space saving was -- 50MB.

Until we do a separate build image, I don't think this is worth optimising further. It only became less readable.

**Pull request merged**